data-structures/data.md

```diff
@@ -0,0 +1,28 @@
+`Flexible Byte Layout` is an advanced layout for representing binary data.
+
+It is flexible enough to support very small and very large (multi-block) binary data.
+
+```sh
+type NestedByte struct {
+  lengthHint Int
+  part &FlexibleByteList
+} representation tuple
+
+type NestedByteList [ NestedByte ]
```

Comment:
The singular ( one element ) is clashing cognitively with "single byte". Perhaps  `BytesSegment` ?